sysconfig/cloudinit.go

```diff
@@ -28,8 +28,12 @@ import (
 	"github.com/snapcore/snapd/osutil"
 )
 
+func ubuntuDataCloudDir(rootdir string) string {
+	return filepath.Join(rootdir, writableDefaultsDir, "etc/cloud/")
+}
+
```

Comment:
I've added a more generic helper in #8533 per Samuele's comment under my PR, we can tweak this later depending on which PR lands first.